plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/SQLParserContext.java

```diff
@@ -48,6 +55,19 @@ public class SQLParserContext {
 
     public SQLParserContext(@Nullable DBPDataSource dataSource, @NotNull SQLSyntaxManager syntaxManager, @NotNull SQLRuleManager ruleManager, @NotNull IDocument document) {
         this.dataSource = dataSource;
+        if (dataSource != null) {
```

Comment:
Maybe we should pass only container here. And parsercontext will get datasource from it? Does it make sense to have 2 variables?

Reply:
I agree with this comment

Reply:
it doesn't work in that way. This change will bring a lot of changes in other places so we already decided to not to do that